packages/contracts-bedrock/test/L1/SystemConfig.t.sol

```diff
@@ -226,7 +228,140 @@ contract SystemConfig_Init_CustomGasToken is SystemConfig_Init {
         assertEq(systemConfig.gasPayingTokenSymbol(), token.symbol());
     }
 
-    function test_initialize_customGasToken_fails() external {
+    /// @dev Tests that initialization sets the correct values and getters work.
+    function testFuzz_initialize_customGasToken_succeeds(
+        address _token,
+        string calldata _name,
+        string calldata _symbol
+    )
+        external
+    {
+        vm.assume(bytes(_name).length <= 32);
+        vm.assume(bytes(_symbol).length <= 32);
+
+        // Wipe out the initialized slot so the proxy can be initialized again
+        vm.store(address(systemConfig), bytes32(0), bytes32(0));
+        vm.store(address(systemConfig), GasPayingToken.GAS_PAYING_TOKEN_SLOT, bytes32(0));
+        vm.store(address(systemConfig), GasPayingToken.GAS_PAYING_TOKEN_NAME_SLOT, bytes32(0));
+        vm.store(address(systemConfig), GasPayingToken.GAS_PAYING_TOKEN_SYMBOL_SLOT, bytes32(0));
+
+        vm.mockCall(_token, abi.encodeWithSelector(token.decimals.selector), abi.encode(18));
+        vm.mockCall(_token, abi.encodeWithSelector(token.name.selector), abi.encode(_name));
+        vm.mockCall(_token, abi.encodeWithSelector(token.symbol.selector), abi.encode(_symbol));
+
+        token = ERC20(_token);
+
+        vm.prank(systemConfig.owner());
+        systemConfig.initialize({
+            _owner: alice,
+            _overhead: 2100,
+            _scalar: 1000000,
+            _batcherHash: bytes32(hex"abcd"),
+            _gasLimit: 30_000_000,
+            _unsafeBlockSigner: address(1),
+            _config: Constants.DEFAULT_RESOURCE_CONFIG(),
+            _batchInbox: address(0),
+            _addresses: SystemConfig.Addresses({
+                l1CrossDomainMessenger: address(0),
+                l1ERC721Bridge: address(0),
+                l1StandardBridge: address(0),
+                l2OutputOracle: address(0),
+                optimismPortal: address(optimismPortal),
+                optimismMintableERC20Factory: address(0),
+                gasPayingToken: address(token)
+            })
+        });
+
+        (address addr, uint8 decimals) = systemConfig.gasPayingToken();
+        assertEq(decimals, 18);
+
+        if (_token == address(0) || _token == Constants.ETHER) {
+            assertEq(addr, Constants.ETHER);
+            assertEq(systemConfig.gasPayingTokenName(), "Ether");
+            assertEq(systemConfig.gasPayingTokenSymbol(), "ETH");
+        } else {
+            assertEq(addr, _token);
+            assertEq(systemConfig.gasPayingTokenName(), _name);
+            assertEq(systemConfig.gasPayingTokenSymbol(), _symbol);
+        }
+    }
+
+    /// @dev Tests that initialization sets the correct values and getters work when token address passed is 0.
+    function test_initialize_customGasToken_zeroTokenAddress_succeeds() external {
+        // Wipe out the initialized slot so the proxy can be initialized again
+        vm.store(address(systemConfig), bytes32(0), bytes32(0));
```

Comment:
we could dedupe the cleaning the storage + call to init in an internal function possibly?

Reply:
good idea!

Reply:
check now!